Sources/SafeDICore/Generators/ScopeGenerator.swift

```diff
@@ -169,22 +177,36 @@ actor ScopeGenerator: CustomStringConvertible {
                         let functionArguments = forwardedProperties.isEmpty ? "" : forwardedProperties.initializerFunctionParameters.map(\.description).joined()
                         let functionName = self.functionName(toBuild: property)
                         let functionDeclaration = isPropertyCycle ? "" : """
-                            func \(functionName)(\(functionArguments)) -> \(instantiable.concreteInstantiableType.asSource) {
+                            func \(functionName)(\(functionArguments)) -> \(concreteTypeName) {
                             \(generatedProperties.joined(separator: "\n"))
                             \(Self.standardIndent)\(generatedProperties.isEmpty ? "" : "return ")\(returnLineSansReturn)
                             }
 
                             """
 
                         let typeDescription = property.typeDescription.asSource
-                        let unwrappedTypeDescription = property.typeDescription.unwrappedTypeDescription.asSource
+                        let unwrappedTypeDescription = property
+                            .typeDescription
+                            .unwrappedTypeDescription
+                            .asSource
+                        let instantiatedTypeDescription = property
+                            .typeDescription
+                            .unwrappedTypeDescription
+                            .asInstantiatedType
+                            .asSource
                         let propertyDeclaration = if typeDescription == unwrappedTypeDescription {
                             "let \(property.label)"
                         } else {
-                            "let \(property.label): \(typeDescription)"
+                            "let \(property.asSource)"
```

Comment:
not a functional change, but this change makes our code generation a little more consistent / less repetitive.